### DISCUSSION TITLE
Change Chart API /api/v1/chart/{pk}

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#22256] [Q&A / Help] [answered]
Need some help with the API endpoint `/api/v1/chart/{pk}` to change/update the chart.
I am trying out the API from `/swagger/v1` UI, and I am able to perform `GET` request on the above mentioned endpoint which returns the chart information. But at same time, `POST` request to the same chart fails with below error.

```
{
  "errors": [
    {
      "message": "400 Bad Request: The CSRF token is missing.",
      "error_type": "GENERIC_BACKEND_ERROR",
      "level": "error",
      "extra": {
        "issue_codes": [
          {
            "code": 1011,
            "message": "Issue 1011 - Superset encountered an unexpected error."
          }
        ]
      }
    }
  ]
}
```
Please share your opinion.

**Comment by rusackas** [comment, marked as answer]
You either need to provide a CSRF token along with the auth token, or (for local testing purposes, where security is not an issue) you can set `WTF_CSRF_ENABLED = False` in the config.py

**Reply by kamalkeshavani-aiinside** [reply]
I am currently trying with swagger UI. This is the schema seen in swagger UI for this endpoint. Where should I include the CSRF token? Or should it go in request headers like access_token, then can I test that in swagger UI?
```
{
  "cache_timeout": 0,
  "certification_details": "string",
  "certified_by": "string",
  "dashboards": [
    0
  ],
  "datasource_id": 0,
  "datasource_type": "druid",
  "description": "string",
  "owners": [
    0
  ],
  "params": "string",
  "query_context": "string",
  "query_context_generation": true,
  "slice_name": "string",
  "viz_type": [
    "bar",
    "line_multi",
    "area",
    "table"
  ]
}
```

**Reply by kamalkeshavani-aiinside** [reply]
@rusackas

**Reply by rusackas** [reply]
Yes, I believe you add it as `x-csrftoken` in the headers with the request. Or disable the need for it if this is local experimentation.

**Reply by kamalkeshavani-aiinside** [reply]
@rusackas Thank you. I am able to use it with `x-csrftoken` header.
